presto-spi/src/main/java/com/facebook/presto/spi/function/FunctionDescriptor.java

```diff
@@ -0,0 +1,170 @@
+     * If function does not transform the shape of the input, then all collected subfields accessed from lambdas from outer calls could be passed to the input column or next
+     * nested call. If function does change the input then the transformation could be supplied in <code>outerLambdaSubfieldsTransform</code>.
+     * If <code>isTransformingInput=true</code> and <code>outerLambdaSubfieldsTransform</code> is supplied, then all the accessed subfields from outer calls will be transformed
+     * using the transformation function in <code>outerLambdaSubfieldsTransform</code> before passing it to the input.
+     * If <code>isTransformingInput=true</code> and <code>outerLambdaSubfieldsTransform</code> is not supplied, then all the accessed subfields from outer calls will be
+     * discarded.
+     * Example. For the table <code>CREATE TABLE (events array(row(id BIGINT, data VARCHAR)))</code>, let's consider the query
+     * <code>SELECT FILTER(TRANSFORM(events, x -> CAST(x AS ROW(key BIGINT, value VARCHAR), y -> y.key > 1000)))</code>.
+     * Here, the subfield <code>key</code> was captured in the function FILTER. And the function TRANSFORM transforms the input that make it impossible to pass the accessed
+     * subfield <code>key</code> from outer call FILTER without transforming the name of the subfield from <code>key</code> to <code>id</code>.
+     */
+    private final boolean isTransformingInput;
+
+    /**
+     * Indicates whether the function prunes the subfields. The function prunes the subfields if and only if it does not output any subfield besides those that this
+     * function and all the nested function accessed.
+     * Example: FILTER function does not prune the subfields while ALL_MATCH function does.
+     */
+    private final boolean isPruningSubfields;
+
+    /**
+     * Set of indices of the CallExpression arguments to that all accessed subfields from the outer CallExpressions relate. Lambda subfields from outer calls will be
+     * passed only to the argument in this set while traversing the expression tree with SubfieldExtractor visitor.
+     * If <code>callArgumentsToPassOuterLambdaSubfields</code> is <code>null</code>, SubfieldExtractor will pass outer lambda subfields to all the arguments of the
+     * CallExpression associated with the current function.
+     * If <code>callArgumentsToPassOuterLambdaSubfields</code> is <code>ImmutableSet.of()</code>, SubfieldExtractor will not pass outer lambda subfields to any
+     * of the CallExpression arguments associated with the current function.
+     */
+    private final Set<Integer> callArgumentsToPassOuterLambdaSubfields;
+
+    /**
+     * Contains the transformation function that needs to be applied to the lambda subfield from outer functions.
+     * This parameter could be non-null only if function semantic requires the changes in the lambda subfields collected from outer functions and therefore
```

Comment:
I would just say - This field will be present only if isTransformingInput is true.

Reply:
Added change